andela/ah-kgl-avengers-backend
# #165020212 Like, Dislike and Favourite an Article

## What does this PR do?
- creates the endpoint for the user to like or dislike the article.
- creates the endpoint for the user to favorite an article and can read again.
- implement the functionality to return the number of likes that the article has.

## Description of Task to be completed?
- Implement the like and dislike of article functionalities.
- Implement the functionality for returning the number of likes in the article.
- Implementing the functionality for the favoriting article and get the all favorited articles.

## How should this be manually tested?
Set up the project following the README.md.
Using an API client like Postman or Insomnia:
- You first need to signup by Making a POST request to localhost:3000/api/v1/auth/signup with a JSON body that includes username,email and password fields with corresponding values:
{ "username": "username", email : your.email@address.com, "password": YourPassword }
You should get a 201 OK response if all data are valid
- Second You will require to login and get the token to be authenticated by Making a POST request to localhost:3000/api/v1/auth/login with a JSON body that includes email and password fields with corresponding values:
{ "email" : your.email@address.com, "password": YourPassword }
You should get a 200 OK response if all data are valid.
- To test the like endpoint Make a POST request to `localhost:3000/api/v1/articles/:slug/like` with a token in headers: 
{ "status": 200, `message`: `You have successfully liked this article` }
 You should get a `200 OK` response if all data are valid.
- To test the remove like endpoint Make a POST request to `localhost:3000/api/v1/articles/:slug/like` with a token in headers, the second time: 
{ "status": 200, `message`: `you have successfully removed your like` }
 You should get a `200 OK` response if all data are valid.

- To test the dislike endpoint Make a DELETE request to `localhost:3000/api/v1/articles/:slug/dislike` with a token in headers: 
{ "status": 200, `message`: `You have successfully disliked this article` }
 You should get a `200 OK` response if all data are valid.
- To test the remove dislike endpoint Make a POST request to `localhost:3000/api/v1/articles/:slug/dislike` with a token in headers, the second time: 
{ "status": 200, `message`: `you have successfully removed your dislike` }
 You should get a `200 OK` response if all data are valid.

- To test the favorite endpoint Make a POST request to `localhost:3000/api/v1/articles/:slug/favorite`, You should get a `200 OK` response and the Object { "status": 200, `message`: `You have favorited this article` } and To remove the favorite you can make a request twice and you will get a `200 oK` response with object { "status": 200, `message`: `You have successfully removed your favorite` }.

- To test the return of the number of likes on article make a GET request on localhost:3000/api/v1/articles/:slag and you will get the response of all the information of the article including the number of likes.

- To test the return of all favorited article smake a GET request on localhost:3000/api/v1/users/favorite and you will get the response of all the articles you have favorited.
## Any background context you want to provide?
You will need to set up a .env file containing the Facebook and Google OAuth Client IDs, tokens, Cloud name, Cloud API KEY, and CLOUD API SECRET.
## What are the relevant pivotal tracker stories?
[165020212](https://www.pivotaltracker.com/n/projects/2321840)
